### PR TITLE
Added EntityData.nitrox to zip backup

### PR DESCRIPTION
**Pull request by ProjectPatatoe**
EntityData was not being added in backup zip

### CODE REVIEW

**Review by dartasen — APPROVED** [review]

**Comment by dartasen** [comment]
Maybe It should be better to add every file inside the save dir, so we don't need to maintain this area everytime the save format is changed

**Comment by ProjectPatatoe** [comment]
yea thats a good idea. I'll change that.

**Comment by ProjectPatatoe** [comment]
I made it for any .nitrox because an existing zip would be included. If we want to change where the zip is saved then it would make sense to grab the entire directory.

**Review by Coding-Hen — APPROVED** [review]

File: NitroxServer/Serialization/World/WorldPersistence.cs [hunk 1]
```diff
@@ -124,10 +124,8 @@ internal Optional<World> LoadFromFile(string saveDir)
                 //Backup world if loading fails
                 using (ZipFile zipFile = new ZipFile())
                 {
-                    zipFile.AddFile(Path.Combine(saveDir, "Version" + fileEnding));
-                    zipFile.AddFile(Path.Combine(saveDir, "BaseData" + fileEnding));
-                    zipFile.AddFile(Path.Combine(saveDir, "PlayerData" + fileEnding));
-                    zipFile.AddFile(Path.Combine(saveDir, "WorldData" + fileEnding));
+                    string[] nitroxFiles = Directory.GetFiles(saveDir, "*.nitrox");
```
Comment (Jannify) [inline NitroxServer/Serialization/World/WorldPersistence.cs:127]: That doesn't include json files if used (see `fileEnding`).

**Review by Jannify — CHANGES_REQUESTED** [review]

**Review by Jannify — APPROVED** [review]

**Review by dartasen — APPROVED** [review]

**Review by Coding-Hen — APPROVED** [review]

**Review by Measurity — APPROVED** [review]

**Pull request merged**